src/controllers/__tests__/integrationTokens.spec.ts

```diff
@@ -0,0 +1,356 @@
+    (mockCache.adapter.get as jest.Mock).mockResolvedValue(hashedToken);
+
+    // Simulate a fatal register failure
+    (mockedUser.register as jest.Mock).mockResolvedValue(null);
+
+    const resp = await request(app)
+      .post('/apollo-signup')
+      .set('X-CSRF-Token', csrfToken)
+      .set('Accept', 'application/json')
+      .set('Content-Type', 'application/json')
+      .send(JSON.stringify(mockedUserData));
+
+    expect(resp.statusCode).toEqual(500);
+    expect(resp.headers['x-csrf-token']).toBeTruthy();
+    expect(resp.headers['set-cookie']).toBeFalsy();
+    expect(resp.body).toEqual({ success: false, message: 'Unable to register the account.' });
+  });
+
+  it('POST /apollo-signin should generate access token and refresh token cookies on success', async () => {
+    const hashedToken = createHash('sha256').update(`${csrfToken}${generalConfig.hashTokenSecret}`).digest('hex');
+    (mockCache.adapter.get as jest.Mock).mockResolvedValue(hashedToken);
+
+    const registeredUser = mockedUser;
+    registeredUser.id = casual.integer(1, 999);
+    (mockedUser.login as jest.Mock).mockResolvedValue(registeredUser);
+
+    const resp = await request(app)
+      .post('/apollo-signin')
+      .set('X-CSRF-Token', csrfToken)
+      .set('Accept', 'application/json')
+      .set('Content-Type', 'application/json')
+      .send(JSON.stringify({ email: mockUser.email, password: mockUser.password }));
+
+    expect(resp.statusCode).toEqual(200);
+    expect(resp.headers['x-csrf-token']).toBeTruthy();
+    const cookies = processResponseCookies(resp.headers);
+    expect(cookies['dmspt']).toBeTruthy();
+    expect(cookies['dmspr']).toBeTruthy();
+    expect(resp.body).toEqual({ success: true, message: 'ok' });
+
+    // Make sure the cache contains the refresh tokens
+    const cachedToken = Object.keys(mockRedis).find((key) => { return key.includes(`dmspr:`) });
+    expect(cachedToken).toBeTruthy();
+  });
+
+  it('POST /apollo-signin should NOT generate access token and refresh token cookies on failure', async () => {
+    const hashedToken = createHash('sha256').update(`${csrfToken}${generalConfig.hashTokenSecret}`).digest('hex');
+    (mockCache.adapter.get as jest.Mock).mockResolvedValue(hashedToken);
+
+    // Simulate a fatal register failure
+    (mockedUser.login as jest.Mock).mockResolvedValue(null);
+
+    const resp = await request(app)
+      .post('/apollo-signin')
+      .set('X-CSRF-Token', csrfToken)
+      .set('Accept', 'application/json')
+      .set('Content-Type', 'application/json')
+      .send(JSON.stringify(mockedUserData));
+
+    expect(resp.statusCode).toEqual(401);
+    expect(resp.headers['x-csrf-token']).toBeTruthy();
+    expect(resp.headers['set-cookie']).toBeFalsy();
+    expect(resp.body).toEqual({ success: false, message: 'Invalid credentials' });
+  });
+
+  it.only('POST /apollo-refresh should refresh the tokens', async () => {
```

Comment:
L258: Only this unit test is running because of the "it.only". I removed it and tried running the unit test and the last test failed.

Reply:
ok thanks. I will take a look